docs/specs/openapi.yaml

```
@@ -0,0 +1,335 @@
+            Optional timeout duration.
+            If specified, wait till change is ready or timeout occurs, whichever is first.
+            Defaults to zero if not specified, in which case the request will wait indefinitely until the change is ready.
+            Read more on the format of the timeout [here](#timeout).
+      responses:
+        '200':
+          description: Change information.
+          content:
+            application/json:
+              schema:
+                type: object
+                properties:
+                  type:
+                    type: string
+                    enum: ["sync"]
+                  status-code:
+                    type: integer
+                    example: 200
+                  status:
+                    type: string
+                    example: "OK"
+                  result:
+                    $ref: '#/components/schemas/ChangeInfo'
+              examples:
+                start service:
+                  value:
+                    type: sync
+                    status-code: 200
+                    status: OK
+                    result:
+                      id: "6"
+                      kind: "start"
+                      summary: "Start service \"svc1\""
+                      status: "Done"
+                      tasks:
+                        - id: "4"
+                          kind: "start"
+                          summary: "Start service \"svc1\""
+                          status: "Done"
+                          progress:
+                            label: ""
+                            done: 1
+                            total: 1
+                          spawn-time: "2024-12-12T12:50:29.743512342+08:00"
+                          ready-time: "2024-12-12T12:50:30.751092759+08:00"
+                      ready: true
+                      spawn-time: "2024-12-12T12:50:29.743570175+08:00"
+                      ready-time: "2024-12-12T12:50:30.751094634+08:00"
+        '400':
+          description: Bad Request - invalid timeout parameter.
+          content:
+            application/json:
+              schema:
+                $ref: "#/components/schemas/ErrorResponse"
+        '404':
+          description: Not Found - change ID not found.
+          content:
+            application/json:
+              schema:
+                $ref: "#/components/schemas/ErrorResponse"
+        '504':
+          description: Gateway Timeout - timeout waiting for change.
+          content:
+            application/json:
+              schema:
+                $ref: "#/components/schemas/ErrorResponse"
+  /v1/services:
+    get:
+      summary: Get services
+      description: Fetch information about specific services (or all of them), ordered by service name.
+      tags:
+        - services
+      parameters:
+        - in: query
+          name: names
+          description: The names of the services to get, a comma-separated string. If empty, get all services.
```

I believe this supports both comma-separated and multiple parameters. Is there a way to specify "this parameter can occur multiple times" in OAS? If so, let's use that. In any case, let's not mention the comma-separated version as I think it's better to let HTTP do that, so something like:

```suggestion
          description: The names of the services to get. To get multiple services, specify this parameter multiple times. If not set, get all services.
```